apps/whale-api/src/module.api/poolpair.controller.e2e.ts

```diff
@@ -527,11 +535,84 @@ describe('get best path', () => {
           poolPairId: '23',
           priceRatio: { ab: '0.14285714', ba: '7.00000000' },
           tokenA: { id: '10', name: 'J', symbol: 'J', displaySymbol: 'dJ' },
-          tokenB: { id: '11', name: 'K', symbol: 'K', displaySymbol: 'dK' }
+          tokenB: { id: '11', name: 'K', symbol: 'K', displaySymbol: 'dK' },
+          commissionFeeInPct: '0.25000000'
         }
       ],
       estimatedReturn: '7.00000000',
-      estimatedReturnLessDexFees: '7.00000000'
+      estimatedReturnLessDexFees: '5.25000000'
+    })
+  })
+
+  it('should deduct commission fee - 1 leg', async () => {
+    const response = await controller.getBestPath('10', '12')
+    expect(response).toStrictEqual({
+      fromToken: {
+        id: '10',
+        name: 'J',
+        symbol: 'J',
+        displaySymbol: 'dJ'
+      },
+      toToken: {
+        id: '12',
+        name: 'L',
+        symbol: 'L',
+        displaySymbol: 'dL'
+      },
+      bestPath: [
+        {
+          symbol: 'J-L',
+          poolPairId: '24',
+          priceRatio: { ab: '0.50000000', ba: '2.00000000' },
+          tokenA: { id: '10', name: 'J', symbol: 'J', displaySymbol: 'dJ' },
+          tokenB: { id: '12', name: 'L', symbol: 'L', displaySymbol: 'dL' },
+          commissionFeeInPct: '0.10000000'
+        }
+      ],
+      estimatedReturn: '2.00000000',
+      estimatedReturnLessDexFees: '1.80000000'
+    })
+  })
+
+  it('should deduct commission and dex fees - 2 legs', async () => {
+    const response = await controller.getBestPath('10', '13')
+    expect(response).toStrictEqual({
+      fromToken: {
+        id: '10',
+        name: 'J',
+        symbol: 'J',
+        displaySymbol: 'dJ'
+      },
+      toToken: {
+        id: '13',
+        name: 'M',
+        symbol: 'M',
+        displaySymbol: 'dM'
+      },
+      bestPath: [
+        {
+          symbol: 'J-L',
+          poolPairId: '24',
+          priceRatio: { ab: '0.50000000', ba: '2.00000000' },
+          tokenA: { id: '10', name: 'J', symbol: 'J', displaySymbol: 'dJ' },
+          tokenB: { id: '12', name: 'L', symbol: 'L', displaySymbol: 'dL' },
+          commissionFeeInPct: '0.10000000'
+        },
+        {
+          symbol: 'L-M',
+          poolPairId: '26',
+          priceRatio: { ab: '0.12500000', ba: '8.00000000' },
+          tokenA: { id: '12', name: 'L', symbol: 'L', displaySymbol: 'dL' },
+          tokenB: { id: '13', name: 'M', symbol: 'M', displaySymbol: 'dM' },
+          commissionFeeInPct: '0.50000000',
+          estimatedDexFeesInPct: {
+            ab: '0.09000000',
+            ba: '0.07000000'
+          }
+        }
+      ],
+      estimatedReturn: '16.00000000',
+      estimatedReturnLessDexFees: '6.09336000'
```

Comment:
Can we have comments explaining which figures were used to calculate this final value?